app/components/thumbnail.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+const MAX_THUMBNAIL_DIMENSION = 999;
+
+export default class Thumbnail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      failed: false,
+    };
+  }
+
+  getThumbnailSrc({ origin, width, height, src }) {
+    const srcPath = src.split('//').pop();
+    return (`${origin}/${width}x${height}/${srcPath}`);
+  }
+
+  handleError() {
+    if (!this.state.failed) {
```

Comment:
That's from what I mentioned here: https://github.com/zooniverse/Panoptes-Front-End/pull/3330#discussion_r92707937